tensorleap/helm-charts
# Add tensorleap-infra chart with elastic operator and use elastic version 7.17 (support auto-migration)

- Add elastic operator into crds directory
- Update elastic search templates - use eck operator Kind and update the pvc name
- Update default elastic url for node and engine
- Change the manifest - add support for `tensorleap-infra` chart and adding installer and app version to enforce CLI upgrade and reinstall
- Refactor the installation process to support reinstall and the new installation manifest version


This PR uses an `elasticsearch` operator to install elastic - it works on a clean environment but needs more testing on environments with DNS problems

The way we chose to upgrade the data is by first upgrading to 7.17 and then to the latest version 8.8.1

This branch also added support in `reinstall` - since it required support for the new elastic PVC

## Tests
Since this branch refactored the whole installation flow to support the new installation manifest and support reinstall on both `install` and `upgrade` in addition to upgrading the elastic, I did the following test to make sure we don't have any regressions
- Run `Install`, `Upgrade` `Reinstall` commands - and then import projects and train 
- Install older version import all the projects(7) with old CLI and then run CLI upgrade and make sure the elastic migration worked well, then try to train projects
- Test `create-manifest` and `pack-installation` commands
- Run airgap installation
- Test the reinstall functionality (app version change or installation params changed)
